great_expectations/expectations/expectation.py

```diff
@@ -265,7 +267,7 @@ def wrapper(
 
 
 # noinspection PyMethodParameters
-class MetaExpectation(ABCMeta):
+class MetaExpectation(ModelMetaclass):
```

Comment:
Why do we use `ModelMetaClass` and inherit from `BaseModel`? Are we losing out on any ABC functionality here? Googling `ModelMetaClass` leads me to believe it is not public in the most recent release.

Reply:
+1 that [we probably should not use this](https://github.com/pydantic/pydantic/issues/6381#issuecomment-1619296261) given that we want to upgrade to pydantic v2 some day.

Reply:
Great call out - I'll do some research to see how we can alleviate this concern

Reply:
We already rely on this for FDS datasources.

pydantic v1 is already embedded inside pydantic v2.

If the goal is simply to allow `Expectations` to be a pydantic model I don't see an alternative that doesn't add a lot more complexity.